tpm/tpm.go

```diff
@@ -988,6 +988,35 @@ func GetPermanentFlags(rw io.ReadWriter) (PermanentFlags, error) {
 	return ret, err
 }
 
+// GetAlgs returns a list of algorithms supported by the TPM device.
+func GetAlgs(rw io.ReadWriter) ([]Algorithm, error) {
+	var algs []Algorithm
+	for _, i := range []Algorithm{AlgRSA: AlgXOR} {
```

Comment:
This doesn't do what you think https://play.golang.org/p/7ilCvaudWxz
You're creating a slice of Algorithm, with all values being 0 except for value at index AlgRSA which is set to AlgXOR.
You want:
```
for i := AlgRSA; i <= AlgXOR; i++ {
```

Reply:
I'm really thankful for your patience:)